contracts/WildcardSteward_v3.sol

```diff
@@ -353,13 +353,39 @@ contract WildcardSteward_v3 is Initializable {
     function changeReceivingBenefactor(
         uint256 tokenId,
         address payable _newReceivingBenefactor
-    ) public onlyReceivingBenefactorOrAdmin(tokenId) {
+    )
+        public
+        onlyReceivingBenefactorOrAdmin(tokenId)
+        updateBenefactorBalance(benefactors[tokenId])
+        updateBenefactorBalance(_newReceivingBenefactor)
```

Comment:
Before changing the receiving benefactor, we need to first update the benefactor balance of the old and new benefactors, to sure there t, time last collected is up until now. This allows us to adjust both of their totalBenefecatorNumerator in accordance with the token changing benefactors.

Reply:
makes complete sense